### ISSUE TITLE
[DMR] All call rx group list.

### ISSUE THREAD

**Issue by forkoz** [rogerclarkmelbourne/Radioddity_GD-77#30]
So for point to point DMR. "all call" cannot be added to the RX group list. Not sure if its a radio or CPS limitation. But in theory, you could pick a channel & color and then broadcast + receive to all groups on a slot. This would negate having to set up groups or enabling monitor for radio to radio communication.

**Comment by rogerclarkmelbourne** [comment]
How would you onfigure this in the CPS?

I thought the All call, was an option in the Digital Contacts, which perhaps is not accessible

BTW. on related point... selecting None for the TG for the Channel, causes a lot of confusion, as the radio then can’t receive anything 

I think it would be better if the code that checks the RxGrouo list, also checks the Tx TG and includes that in its acceptance list.
I will raise a separate issue for it.

**Comment by forkoz** [comment]
To set it up in CPS is simple.  Add all call as a contact. But it can't be added to any list. Manually entering 16777215 produces the message: ID out of range. You would kill 2 birds with one stone since I can set the TX contact to "all call"

**Comment by rogerclarkmelbourne** [comment]
When do you get that "ID out of range" error ?

I don't get it when I add the contact to my channel

**Comment by forkoz** [comment]
Add new contact. Leave it at group call and write: 16777215

**Comment by forkoz** [comment]
This contact in RX list would receive all bits, much like monitor mode.

**Comment by rogerclarkmelbourne** [comment]
Why can't you simply add an All Call to the contacts ?

Does it have to be a Group call with that number?

Whats the difference ?

**Comment by forkoz** [comment]
I can add it to contacts but then I can't add it to any group lists. So I can have the radio transmit to all but it will receive none... or only the specific IDs in the list.

**Comment by rogerclarkmelbourne** [comment]
OK.

I can modify the CPS to allow it to be added to the list, and then we can see what the radio does !

**Comment by forkoz** [comment]
You can then "cheat" and just set allcall as the group for those "none" channels.

**Comment by rogerclarkmelbourne** [comment]
OK.
Ideally you should be able just to set a Tx of a specific TG (not an all call), and leave Rx Group as none if you want.

**Comment by forkoz** [comment]
Yep, more the merrier. None is useless, there was a note about it in the chinese firmware changelog.

**Comment by rogerclarkmelbourne** [comment]
Which version number in the changelog?

**Comment by forkoz** [comment]
MD_760_V2.5.9.sgl
1. Solve the problem that the group call list is empty and confirm the call of the group call.
2017-08-24

**Comment by forkoz** [comment]
They may have included the chinese stuff in the radioddity versions. I notice that the IAR project went from MD-760(repair bug) to 760(plus) in the newer v3 etc.

**Comment by rogerclarkmelbourne** [comment]
Umm. It would be possible to compare the binaries and see which function changed.

It may be worthwhile as it would give an indication of which function handles this stuff

**Comment by forkoz** [comment]
Only 1 version was on the site :(

**Comment by rogerclarkmelbourne** [comment]
:-(

**Comment by forkoz** [comment]
With the patch, allcall should be both TX and RX. Its transmitting but I don't have a second radio to test. So modifying the CPS would probably be redundant.

**Comment by rogerclarkmelbourne** [comment]
Ok..

r0 seems to contain only the number of the TG, and not the flags about the call type, so I don’t know what would happen if you used and all call ID

**Comment by forkoz** [comment]
I have all call saved as a contact.  As long as that contact is now the RX group in theory the radio is transmitting and receiving to all.  If you have 2 radios see if it works.

**Comment by rogerclarkmelbourne** [comment]
Ok. I will test later today

**Comment by forkoz** [comment]
Having 2 radios now, I've found the following:
1. All call doesn't do anything to receive.
2. 2 radios on all call TX will talk to eachother with no group list but not pick anything else up.

**Comment by rogerclarkmelbourne** [comment]
Probably the best way to have All call work on receive is via whatever method they used for Monitor mode.

There is probably a spare flag bit we can use in the channel data.

Btw.
If you try my direct TG patch, you can enter a group call of 0 at the moment, this seems to do some things which seem similar to an all call

**Comment by forkoz** [comment]
Yep, I was just waiting on that to not write into the CP. :)
